libvips/foreign/jpegsave.c

```diff
@@ -194,6 +198,13 @@ vips_foreign_save_jpeg_class_init( VipsForeignSaveJpegClass *class )
 		G_STRUCT_OFFSET( VipsForeignSaveJpeg, quant_table ),
 		0, 8, 0 );
 
+    VIPS_ARG_ENUM( class, "subsample_mode", 19,
+            _( "Subsample mode" ),
+            _( "Select chroma subsample operation mode" ),
+            VIPS_ARGUMENT_OPTIONAL_INPUT,
+            G_STRUCT_OFFSET( VipsForeignSaveJpeg, subsample_mode ),
+            VIPS_TYPE_FOREIGN_JPEG_SUBSAMPLE,
+            VIPS_FOREIGN_JPEG_SUBSAMPLE_AUTO );
```

Comment:
I think we should tag `no_subsample` as deprecated. That'll hide it from APIs and docs, but it'll still work from old code.

Reply:
done